crates/oxc_linter/src/rules/jsx_a11y/prefer_tag_over_role.rs

```diff
@@ -0,0 +1,137 @@
+use std::collections::HashMap;
+
+use crate::{context::LintContext, rule::Rule, utils::has_jsx_prop_lowercase, AstNode};
+use oxc_ast::{
+    ast::{JSXAttributeItem, JSXAttributeValue, JSXElementName},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-jsx-a11y/prefer-tag-over-role: Prefer using semantic HTML tags over `role` attribute.")]
+#[diagnostic(
+    severity(warning),
+    help("Replace HTML elements with `role` attribute to corresponding semantic HTML tags.")
+)]
+struct PreferTagOverRoleDiagnostic {
+    pub span: Span,
+    pub tag: String,
+    pub role: String,
+}
+#[derive(Debug, Default, Clone)]
+pub struct PreferTagOverRole;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforces using semantic HTML tags over `role` attribute.
+    ///
+    /// ### Why is this bad?
+    /// Using semantic HTML tags can improve accessibility and readability of the code.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// <div role="button" />
+    ///
+    /// // Good
+    /// <button />
+    ///
```

Comment:
```suggestion
    /// ```
```

Reply:
Thank you for taking the time to do this!
Fixed in [fix: commented out part](https://github.com/oxc-project/oxc/pull/1831/commits/8f0f7c6d707f4871e0119e6d00f2fbc388984f0b)🙏🏻